### PR TITLE
Allow move across file systems (Issue 1411)

### PR DESCRIPTION
**Pull request by ArniDagur**
See #1411

### CODE REVIEW

**Comment by wgjak47** [comment]
`fs::copy` didn't work for directory, so I try to call `copy` function in `cp` crate instead....
https://github.com/uutils/coreutils/pull/1416

**Comment by Luke-Nukem** [comment]
Hmm, this is a bit urgent for me - I can't use uutils while packaging for openSUSE due to osc creating files in /tmp before moving to work dir.

File: src/mv/mv.rs [hunk 1]
```diff
@@ -393,6 +404,75 @@ fn rename(from: &PathBuf, to: &PathBuf, b: &Behaviour) -> Result<()> {
     Ok(())
 }
 
+/// A wrapper around `fs::rename`, so that if it fails, we try falling back on
+/// copying and removing.
+fn rename_with_fallback(from: &PathBuf, to: &PathBuf) -> io::Result<()> {
+    if fs::rename(from, to).is_err() {
```
Comment (Arcterus) [inline src/mv/mv.rs]: Is there a specific error for this problem?

**Comment by Arcterus** [comment]
Does this solve the problem fixed by #1416 now?  I notice that you updated the code after @wgjak47's comment.

**Comment by sylvestre** [comment]
@ArniDagur ping?

**Comment by ArniDagur** [comment]
@sylvestre Yeah, this PR was meant to fix #1416, and it did so last time I tested it. I no longer have a computer with more than one file system, however, so maybe it'd be best if you tried it out yourself.

**Comment by sylvestre** [comment]
@ArniDagur Can you allow me to modify your PR ?
it is conflicting, thanks

**Comment by ArniDagur** [comment]
@sylvestre Be my guest!

**Comment by sylvestre** [comment]
For some reasons, I am not able to update your PR.
I created https://github.com/uutils/coreutils/pull/1524 and keep your credit

**Pull request closed without merging**